File: CodeForces/GlobalRound2/a.py
def go():
    n = int(input())
    a = [int(i) for i in input().split(' ')]
    d = {}
    for i in range(n):
        d.setdefault(a[i], [])
        d[a[i]].append(i)
    k = 0
    for i in d:
        for j in d:
            if i != j:
                k = max(k, abs(d[i][0] - d[j][-1]))
    return k

print(go())

File: CodeForces/GlobalRound2/a.c
#include <stdio.h>

int max(int a, int b) {
    if (a > b) {
        return a;
    }
    return b;
}

int main() {
    int n, d = 0;
    scanf("%d", &n);
    int a[n + 1];
    for(int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }
    for(int i = 0; i < n; i++) {
        for(int j = i + 1; j < n; j++) {
            if (a[i] != a[j]) {
                d = max(j - i, d);
            }
        }
    }
    printf("%d", d);
    return 0;
}
